### PR TITLE
feat!: Add support for WASM operations

### PR DESCRIPTION
**Pull request by johnchildren**
BREAKING CHANGE: Added a `number_of_ws` field to `SerialCircuit`

### CODE REVIEW

File: src/circuit_json.rs [hunk 1]
```diff
@@ -137,6 +137,17 @@ pub enum Classical {
     },
 }
 
+/// Additional fields for Wasm operations.
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
+pub struct Wasm {
+    n: u32,
```
Comment (johnchildren) [inline src/circuit_json.rs:143]: probably should be u64

**Comment by johnchildren** [comment]
> Where is this part of the schema defined? It's not in https://github.com/CQCL/tket/blob/develop/schemas/circuit_v1.json

A very good question... I had to dig into the C++ to find out how it worked: https://github.com/CQCL/tket/blob/da1f491059b608701701da6202abb96c67ac6031/tket/src/Ops/ClassicalOps.cpp#L143

**Comment by aborgna-q** [comment]
-.-

I'll make an issue

**Comment by johnchildren** [comment]
Honestly if they aren't too set on the schema yet there are some improvements I'd perhaps like to make such as renaming `wasm_file_uid` to `uid`

**Review by aborgna-q — APPROVED** [review]
LGTM

We can wait for https://github.com/CQCL/tket/issues/1505 to be resolved if you prefer so.

**Comment by johnchildren** [comment]
> LGTM
> 
> We can wait for [CQCL/tket#1505](https://github.com/CQCL/tket/issues/1505) to be resolved if you prefer so.

Yeah I guess I'd just want to hear what the outcome of that is first, happy to wait a little while as I don't need this till later in the week I expect.

**Comment by aborgna-q** [comment]
Merging this as-is. We can update it once the tket issue is resolved.

**Pull request merged**